src/printer.rs

```diff
@@ -263,31 +286,57 @@ impl<W: WriteColor> Printer<W> {
             self.write(b":");
         }
         if self.replace.is_some() {
-            let line = re.replace_all(
-                &buf[start..end], &**self.replace.as_ref().unwrap());
+            let mut count = 0;
+            let line = {
+                let replacer = CountingReplacer { count: &mut count, replace : &**self.replace.as_ref().unwrap() };
+                re.replace_all(&buf[start..end], replacer)
+            };
+            if let Some(max_columns) = self.max_columns {
+                if line.len() > max_columns {
+                    let _ = self.wtr.set_color(self.colors.line());
```

Comment:
I think I would rather see this use the match color instead of the line number color.